### PR TITLE
Improve indexer

### PR DESCRIPTION
**Pull request by aenario**
This PR extends and fix tests for #210

### CODE REVIEW

File: tests/index_tests.coffee [hunk 1]
```diff
@@ -1,11 +1,13 @@
 should = require('chai').Should()
 async = require 'async'
 helpers = require './helpers'
+indexer = require '../server/lib/indexer'
 
 # connection to DB for "hand work"
 db = require("#{helpers.prefix}server/helpers/db_connect_helper").db_connect()
 
 client = helpers.getClient()
+indexer = require('../server/lib/indexer')
```
Comment (nono) [inline tests/index_tests.coffee]: `indexer` is required twice in this file

File: tests/index_tests.coffee [hunk 2]
```diff
@@ -178,10 +181,10 @@ describe "Indexation", ->
 
     describe "Reindexing", ->
 
-        it "When I erase the index", (done) ->
+        it "When I stop the DS", helpers.stopApp
```
Comment (nono) [inline tests/index_tests.coffee]: It looks more like a `before` than an `it`

File: tests/index_tests.coffee [hunk 3]
```diff
@@ -46,7 +48,8 @@ describe "Indexation", ->
     before helpers.clearDB db
     before helpers.startApp
 
-    before require('../server/lib/indexer').cleanup
+    before (cb) -> indexer.initialize cb
+    before (cb) -> indexer.cleanup cb
```
Comment (nono) [inline tests/index_tests.coffee]: Don't `before indexer.initialize` and `before indexer.cleanup` work and be enough here?

File: tests/index_tests.coffee (hunk 2 above)
Comment (aenario) [inline tests/index_tests.coffee]: The goal was to have the test explicit in description, but you're right, the stop could be a `before `

**Comment by aenario** [comment]
Thanks @nono for the review

File: tests/index_tests.coffee (hunk 2 above)
Comment (nono) [inline tests/index_tests.coffee]: You can have a description on a `before` or `after`. For an example: https://github.com/cozy-labs/cozy-desktop/blob/master/test/unit/local/index.coffee#L17-L24

**Comment by frankrousseau** [comment]
Thx!

**Pull request merged**